backend/models/friend.js

```diff
@@ -0,0 +1,14 @@
+import { DataTypes } from "sequelize";
+import { sequelize } from "../database.js";
+import { User } from "./user.js";
+
+export const Friend = sequelize.define("Friend", {
+    id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true
+    }
+});
+
+User.belongsToMany(User, { through: Friend, as: "user1", foreignKey: "UserId1" });
+User.belongsToMany(User, { through: Friend, as: "user2", foreignKey: "UserId2" });
```

Comment:
just wondering, do you need both lines? 

also, what if instead of creating a separate Friend model, you add something like this to the User model:  `User.belongsToMany(User, { through: "Friends", as: "friends" })`, does that behave the same?

Reply:
I believe so since its a relationship between two identical models so we have to differentiate one from the other.